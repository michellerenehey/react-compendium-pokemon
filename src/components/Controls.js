import '../styles/Controls.css';

export default function Controls({ query, setQuery, setLoading, order, setOrder }) {
  return (
    <div>
      <input
        type="text"
        placeholder="search poke here"
        value={query}
        onChange={(e) => {
          setQuery(e.target.value);
        }}
      />
      <select value={order} onChange={(e) => setOrder(e.target.value)}>
        <option value="asc">A - Z</option>
        <option value="desc">Z - A</option>
      </select>
      <button onClick={() => setLoading(true)}>Search</button>
    </div>
  );
}
