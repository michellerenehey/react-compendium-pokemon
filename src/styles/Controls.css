.queryClass,
.orderClass,
.searchButtonClass {
  margin: 5px;
  padding: 5px;
  font-family: 'Space Mono', monospace;
  color: green;
  background-color: pink;
  border-radius: 5px;
  border: 2px solid green;
}
