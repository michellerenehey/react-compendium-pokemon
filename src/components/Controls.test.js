import { render, screen } from '@testing-library/react';
import Controls from './Controls';

test('renders a control panel', () => {
  const container = render(
    <Controls
      query="ab"
      setQuery="ab"
      setLoading="false"
      order="asc"
      setOrder="asc"
      types={['fairy']}
      selectedType="fairy"
      setSelectedType="fairy"
    />
  );
  expect(container).toMatchSnapshot();
});
