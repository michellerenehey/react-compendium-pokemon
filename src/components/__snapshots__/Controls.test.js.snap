// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a control panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <input
          class="queryClass"
          placeholder="search poke here"
          type="text"
          value="ab"
        />
        <select
          class="orderClass"
        >
          <option
            value="asc"
          >
            Sort A - Z
          </option>
          <option
            value="desc"
          >
            Sort Z - A
          </option>
        </select>
        <select
          class="typeClass"
        >
          <option
            value="all"
          >
            all
          </option>
          <option
            value="fairy"
          >
            fairy
          </option>
        </select>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-sxix9q-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Search
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <input
        class="queryClass"
        placeholder="search poke here"
        type="text"
        value="ab"
      />
      <select
        class="orderClass"
      >
        <option
          value="asc"
        >
          Sort A - Z
        </option>
        <option
          value="desc"
        >
          Sort Z - A
        </option>
      </select>
      <select
        class="typeClass"
      >
        <option
          value="all"
        >
          all
        </option>
        <option
          value="fairy"
        >
          fairy
        </option>
      </select>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-sxix9q-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Search
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
